File: 14-longest-common-prefix/solution.ts
function longestCommonPrefix(strs: string[]): string {
  if (strs.length === 1) return strs[0];
  strs.sort((a, b) => b.length - a.length);

  for (let i = strs[0].length; i >= 0; i--) {
    const prefix = strs[0].slice(0, i);
    if (strs.every((s) => s.slice(0, i) === prefix)) {
      return prefix;
    }
  }
  return '';
}

File: 14-longest-common-prefix/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans = "";
        int i = 0;
        while(true) {
            char now = '\0';
            for(auto &s : strs) {
                if(s.size() <= i) return ans;
                if(now == '\0') now = s[i];
                else if(now != s[i]) return ans;
            }
            ans += now;
            i++;
        }
    }
};
